staging/src/k8s.io/apimachinery/pkg/util/httpstream/spdy/roundtripper_test.go

```diff
@@ -120,6 +121,24 @@ func localhostCertPool(t *testing.T) *x509.CertPool {
 func TestRoundTripAndNewConnection(t *testing.T) {
 	localhostPool := localhostCertPool(t)
 
+	const (
+		dialerDefault = iota
+		dialerDialFunc
+		dialerDialer
+	)
+	dialerUsed := dialerDefault
+	dialFunc := func(ctx context.Context, network, address string) (net.Conn, error) {
+		dialerUsed = dialerDialFunc
```

Comment:
dialers should be thread safe, maybe you should use soe `atomic...`

Reply:
In this case I don't think it matters as dialing occurs serially due to https://github.com/golang/go/blob/432158b69a50e292b625d08dcfacd0604acbabd3/src/net/dial.go#L451, however it doesn't hurt to add.